pkg/crc/segment/segment.go

```diff
@@ -0,0 +1,82 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/pborman/uuid"
+	"github.com/segmentio/analytics-go"
+)
+
+type Client struct {
+	segmentClient     analytics.Client
+	config            *crcConfig.Config
+	telemetryFilePath string
+}
+
+func NewClient(config *crcConfig.Config, telemetryFilePath string) (*Client, error) {
+	client, err := analytics.NewWithConfig("cvpHsNcmGCJqVzf6YxrSnVlwFSAZaYtp", analytics.Config{
+		DefaultContext: &analytics.Context{
+			App: analytics.AppInfo{
+				Name:    constants.DefaultName,
+				Version: version.GetCRCVersion(),
```

Comment:
Since this is using some values from `constants`/`version`, shouldn't this also hardcode `constants.TelemetryFilePath`? I don't think the calling code cares, and this does not seem to be set to a different value in tests.

Reply:
@cfergeau In the test we are using a mock client with different endpoint, and we can specify different value for this and check the response also if that is the concern here?

Reply:
segment_test.go is not using `NewClient()`, so the 'path' argument is not used by tests so `NewClient()` can be changed to
```
func NewClient(config *crcConfig.Config)  (*Client, error) {
    const telemetryFilePath = filepath.Join(constants.GetHomeDir(), ".redhat", "anonymousId")

...
    return &Client{segmentClient: client, config: config, telemetryFilePath: telemetryFilePath}, nil
}
```
and the constants.go changes can be dropped.

Reply:
@cfergeau Alright I got what you mean now.